### PR TITLE
Simplify docker compose setup

### PR DESCRIPTION
**Pull request by Rubonnek**
## What type of PR is this? (check all applicable)

- [x] Refactor
- [x] Feature
- [ ] Bug Fix
- [ ] Optimization
- [ ] Documentation Update
- [ ] Community Node Submission

## Have you discussed this change with the InvokeAI team?
- [x] Yes - https://github.com/invoke-ai/InvokeAI/pull/5007#discussion_r1378792615
- [ ] No, because: 

      
## Have you updated all relevant documentation?
- [x] Yes
- [ ] No

## Description

Simplify Docker image creation and execution to a single script that spins up the right service in the docker compose file.
## Related Tickets & Documents

- Depends on #5007

## QA Instructions, Screenshots, Recordings
N/A

## Added/updated tests?

- [ ] Yes
- [x] No : same tests should work.

## [optional] Are there any post deployment tasks we need to perform?

Not to my knowledge.

### CODE REVIEW

**Comment by hipsterusername** [comment]
#5007 just merged 🖖

**Review by ebr — COMMENTED** [review]
Looks great - thanks so much for the contribution. Will test on an nvidia system and report back

File: docker/run.sh [hunk 1]
```diff
@@ -1,11 +1,26 @@
...
 
-SCRIPTDIR=$(dirname "${BASH_SOURCE[0]}")
-cd "$SCRIPTDIR" || exit 1
+  local build_args=""
+  local service_name="invokeai-cpu"
 
-docker compose up -d
-docker compose logs -f
+  [[ -f ".env" ]] &&
+    build_args=$(awk '$1 ~ /=[^$]/ {print "--build-arg " $0 " "}' .env) &&
```
Comment (ebr) [inline docker/run.sh:12]: This should avoid commented lines in `.env` - prevents `--build-arg #SOME_VAR=foo`

```suggestion
    build_args=$(awk '$1 ~ /=[^$]/ && $0 !~ /^#/ {print "--build-arg " $0 " "}' .env) &&
```

**Review by ebr — CHANGES_REQUESTED** [review]
Really like the clever solution with YAML anchors!
However, this change breaks the use of standard `docker compose` commands, and we do have users relying on or preferring to use `docker compose` directly. Could we keep the default `docker-compose.yml` with NVIDIA gpu support (the "happy path" for most common use case), while using a `-f docker-compose-custom.yml` (or similar) argument in `run.sh`?
Open to other ideas too. Great work otherwise, thanks for contributing.

File: docker/run.sh [hunk 2]
```diff
@@ -1,11 +1,26 @@
...
+    service_name="invokeai-$(awk -F '=' '/GPU_DRIVER/ {print $2}' .env)"
```
Comment (ebr) [inline docker/run.sh:13]: if `GPU_DRIVER` isn't set in `.env`, we get `service_name=invokeai-` here

**Comment by ebr** [comment]
@Rubonnek works great! TIL about `docker-compose` profiles. Awesome stuff, LGTM

**Review by ebr — APPROVED** [review]
Great work :100:

**Review by hipsterusername — APPROVED** [review]

**Pull request merged**